api/geth_backend.go

```diff
@@ -305,6 +308,90 @@ func (b *GethStatusBackend) startNodeWithAccount(acc multiaccounts.Account, pass
 	return nil
 }
 
+func getKeyFilePathsByAddresses(keyStoreDir string, addresses map[string]struct{}) ([]string, error) {
+	paths := []string{}
+
+	checkFile := func(path string, fileInfo os.FileInfo) error {
+		if fileInfo.IsDir() {
+			return nil
+		}
+
+		rawKeyFile, e := ioutil.ReadFile(path)
+		if e != nil {
+			return fmt.Errorf("invalid account key file: %v", e)
+		}
+
+		var accountKey struct {
+			Address string `json:"address"`
+		}
+		if e := json.Unmarshal(rawKeyFile, &accountKey); e != nil {
+			return fmt.Errorf("failed to read key file: %s", e)
+		}
+
+		address := types.HexToAddress("0x" + accountKey.Address).Hex()
+		if _, ok := addresses[address]; ok {
+			paths = append(paths, path)
+		}
+
+		return nil
+	}
+
+	err := filepath.Walk(keyStoreDir, func(path string, fileInfo os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		return checkFile(path, fileInfo)
+	})
+	if err != nil {
+		return nil, fmt.Errorf("cannot traverse key store folder: %v", err)
+	}
+
+	return paths, nil
+}
+
+func (b *GethStatusBackend) MigrateKeyStoreDir(acc multiaccounts.Account, password, oldDir, newDir string) error {
+	err := b.ensureAppDBOpened(acc, password)
+	if err != nil {
+		return err
+	}
+
+	accountDB := accounts.NewDB(b.appDB)
+	accounts, err := accountDB.GetAccounts()
+	if err != nil {
+		return err
+	}
+
+	settings, err := accountDB.GetSettings()
+	if err != nil {
+		return err
+	}
+
+	addresses := map[string]struct{}{}
+	addresses[settings.EIP1581Address.Hex()] = struct{}{}
+	addresses[settings.WalletRootAddress.Hex()] = struct{}{}
+
+	for _, account := range accounts {
+		addresses[account.Address.Hex()] = struct{}{}
+	}
+
+	paths, err := getKeyFilePathsByAddresses(oldDir, addresses)
+
+	if err != nil {
+		return err
+	}
```

Comment:
Little formatting tweak. It's idiomatic to have the err check directly below the function that generates an error.

```suggestion
	paths, err := getKeyFilePathsByAddresses(oldDir, addresses)
	if err != nil {
		return err
	}
```

Reply:
fixed